### DISCUSSION TITLE
Ikev2 Fragmentation.

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#1601] [Q&A]
I am using strongswan 5.6.0. I need ike fragmentation as described in rfc7383. In IKE_AUTH exchange, responder can not parse reassembled fragments. Is there any bug in strongswan 5.6.0 and I see 174 bytes padding in fragment 1 and  256 byte in fragment 2 responder side. Such this quantity normal? (padding log in encrypted_payload.c file decrypt_content())

**Comment by tobiasbrunner** [comment]
> I am using strongswan 5.6.0.

Any particular reason you use such an old version?

> responder can not parse reassembled fragments.

Who is responder? strongSwan? Where do you observe the weird padding? Any logs?

> Such this quantity normal?

No, padding for IKE messages should be at most 15 (depends on the encryption algorithm). Technically, it could be more (e.g. to mask the actual size of the message, as can be done with ESP packets), but 256 is definitely not possible as the padding length is stored in a single byte (i.e. 255 is the most padding that can be encoded, or did you also count the padding length field?)
